projects/plugins/jetpack/extensions/shared/external-media/sources/with-media.js

```diff
@@ -169,6 +169,11 @@ export default function withMedia( mediaSource = MediaSource.Unknown ) {
 					parse: window.wpcomFetch === undefined,
 				} )
 					.then( result => {
+						// If we don't have media available, we should show an error instead of crashing the editor.
+						if ( result.media === undefined ) {
+							result.code = 'internal_server_error';
+							throw result;
```

Comment:
Throwing the response body seems fine to me, but I wasn't able to validate what the shape of the `error` argument to the `handleApiError` method usually is. Would it "normally" also be the response body?

Reply:
We can't know the exact shape since it's not typesafe. The `handleApiError` uses `code`, `errors`, and `message` properties. The only property that is required is code, the rest are conditionally checked.

I could map the response to: `{ code: 'error code'}` before throwing it.

Reply:
Updated